src/kOS.Safe/Utilities/StringUtil.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+namespace kOS.Safe
+{
+    public static class StringUtil
+    {
+        public static bool EndsWithCall(string str)
+        {
+            int len = str.Length;
+            return len >= 2 && str[len-1] == ')' && str[len-2] == '(';
+        }
+
+        public static bool HasPointerSuffix(string str)
+        {
+            int len = str.Length;
+            return len > 0 && str[len-1] == '*';
+        }
+
+        public static bool HasVarPrefix(string str)
+        {
+            return str.Length > 0 && str[0] == '$';
+        }
+    }
+}
```

Comment:
Should we just have a generic "endswith" and "startswith" call in our utility library that operates on generic strings but does it char by char?  Would it be an improvement over the generic mono one that way?  i.e:
```
public static bool EndsWith(string biggerString, string substring)
{
    int len = subString.Length;
    int i = biggerString.Length - len;
    if (i<0) return false;
    while (i < len)
    {
        if (subSrring[i] != biggerString[j]) // maybe allow for culture case-insensitive comparison here?  Nah, probably not.
            return false;
        ++i;
        ++j;
    }
    return true;
}
```
It seems weird to be writing one hardcoded primitive method per endswith comparison we want to do, but I can conceed it may be worth it if it's faster that the generic solution.